src/OrleansServiceBus/Providers/Streams/EventHub/CachePressureMonitors/SlowConsumingPressureMonitor.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.ServiceBus.Providers
+{
+    /// <summary>
+    /// Pressure monitor which is in favor of the slow consumer in the cache
+    /// </summary>
+    public class SlowConsumingPressureMonitor : ICachePressureMonitor
+    {
+        private static TimeSpan DefaultCheckPeriod = TimeSpan.FromMinutes(1);
+        private const double DefaultFlowControlThreshold = 0.5;
+
+        /// <summary>
+        /// CheckPeriod
+        /// </summary>
+        public TimeSpan CheckPeriod { get; set; }
+        /// <summary>
+        /// FlowControlThreshold
+        /// </summary>
+        public double FlowControlThreshold { get; set; }
+
+        private readonly Logger logger;
+        private double biggestPressureInCurrentPeriod;
+        private DateTime nextCheckedTime;
+        private bool isUnderPressure;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(Logger logger)
+            : this(DefaultFlowControlThreshold, DefaultCheckPeriod, logger)
+        { }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="checkPeriod"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(TimeSpan checkPeriod, Logger logger)
```

Comment:
checkPeriod is a confusing name. It took me some time by reading and re-reading the code to understand what it does, because it's not the time period between checks.
Maybe it should be pressure window size? Or something else not related to check time?